test/integration/buyers/accounts_controller_test.rb

```diff
@@ -391,4 +391,18 @@ class NotLoggedInTest < ActionDispatch::IntegrationTest
       assert_redirected_to provider_login_path
     end
   end
+
+  class ManyAccountsTest < ActionDispatch::IntegrationTest
+    def setup
+      @provider = FactoryBot.create(:provider_account)
+      login! @provider
+    end
+
+    test 'no N+1 problems in the index' do
+      populate = ->(n) {FactoryBot.create_list(:buyer_account, n, provider_account: @provider)}
+      assert_perform_constant_number_of_queries(populate: populate, matching: /users/) do
```

Comment:
do we need the `matching` argument?

Reply:
I think so, I'm using it because the N+1 problem was querying the users table. If I remove it, then it fails b/c the number of queries to the plans table also increments linearly